### ISSUE TITLE
Export redux exports

### ISSUE THREAD

**Issue by jamesplease** [reactjs/react-redux#922]
I think it is a better developer experience when binding libs export the exports from the "core" lib. For instance, right now it is common to see developers import from redux and react-redux in the same component:

```js
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
```

This is extra typing, which makes it a little less convenient to use Redux. Additionally, it can be difficult to remember what comes from which library. I have it memorized now, but I used to always use trial and error to figure out where to import things from.

It would be less code to write, and easier for developers to learn and use redux, if the above was:

```js
import { connect, bindActionCreators } from 'react-redux';
```

React Router does this, and it is 👌:

```js
import { Router, Link } from 'react-router-dom';
```

`Router` comes from the react-router package, and `Link` comes from react-router-dom.

In other words, web users do not need to install react-router and react-router-dom. Redux could have a similar developer experience.

Thanks for reading :v:

**Comment by jamesplease** [comment]
> While convenient, if we want to introduce a new API/export, we have to do so in two places. That seems annoying...

I see what you mean. It would be slightly more work for the maintainers to release two libs instead of one for certain updates, but I think the improvement to the developer experience could be worth the trade-off.

> ...and potentially error-prone

True. This is especially a problem if the people that release redux do not talk to the react-redux team (I'm not too sure about who maintains which project).

If they are different teams who don't talk to one another a lot, then I agree that the chance of errors is a good reason to not make this change.

> Also, React itself doesn't do this between react and react-dom, and it doesn't seem to be a common request over there.

I agree that it doesn't seem to be a common request with react. One explanation for this could be that developers just don't use react-dom exports outside of the index file that mounts the app. Within an individual component, developers will almost exclusively import from the main react package.

This is pretty different when you look at React Router and Redux, where developers commonly use exports from the different npm packages for the libraries within a single component.

Thanks for taking the time to read this issue and to respond @timdorr ! :v:

**Comment by markerikson** [comment]
We maintain both `redux` and `react-redux`, but they _are_ separate packages.

I don't see a real need for this.  Besides, in most cases, you shouldn't have to import things from both libraries in the same file.  If you're using `bindActionCreators` when connecting a React component, I'd encourage you to use the "object shorthand" form of `mapDispatch` instead:

```js
import {addTodo, toggleTodo} from "./todoActions";

export default connect(null, {addTodo, toggleTodo})(TodoList);
```

That binds them automatically.

The only other Redux export that you _might_ be using here is `compose()` for combining multiple HOCs together, and that's a rarer use case.
